src/main/java/athena/task/taskfilter/ForecastFilter.java

```diff
@@ -25,24 +26,45 @@ public ForecastFilter(Forecast forecast) {
      */
     @Override
     public boolean isTaskIncluded(Task task) {
-        boolean isTaskIncluded;
+        for (LocalDate date : task.getDates()) {
+            if (isInstanceIncluded(date)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private static int getWeekNumber(LocalDate taskDate) {
+        TemporalField woy = WeekFields.of(Locale.getDefault()).weekOfWeekBasedYear();
+        return taskDate.get(woy);
+    }
+
+    private boolean isInstanceIncluded(LocalDate taskDate) {
+        boolean isInstanceIncluded;
         if (forecast == Forecast.ALL) {
-            isTaskIncluded = true;
+            isInstanceIncluded = true;
         } else if (forecast == Forecast.WEEK) {
             int currentWeekNumber = getWeekNumber(todayDate);
-            LocalDate taskDate = task.getDate();
             int taskWeekNumber = getWeekNumber(taskDate);
-            isTaskIncluded = (currentWeekNumber == taskWeekNumber);
+            isInstanceIncluded = (currentWeekNumber == taskWeekNumber);
         } else {
-            LocalDate taskDate = task.getDate();
-            isTaskIncluded = taskDate.equals(todayDate);
+            isInstanceIncluded = taskDate.equals(todayDate);
         }
-        return isTaskIncluded;
+        return isInstanceIncluded;
     }
 
-    private static int getWeekNumber(LocalDate taskDate) {
-        TemporalField woy = WeekFields.of(Locale.getDefault()).weekOfWeekBasedYear();
-        return taskDate.get(woy);
+    public Task filterDates(Task task) {
```

Comment:
I think `filterDates` is not very clear. How about something like `removeExcludedDates`?